diff --git a/libi2pd/NetDb.hpp b/libi2pd/NetDb.hpp
index df831be..f739057 100644
--- a/libi2pd/NetDb.hpp
+++ b/libi2pd/NetDb.hpp
@@ -41,12 +41,12 @@ namespace data
 	const int NETDB_MIN_ROUTERS = 90;
 	const int NETDB_MIN_FLOODFILLS = 5;
 	const int NETDB_MIN_TRANSPORTS = 10 ; // otherwise assume offline
-	const int NETDB_NUM_FLOODFILLS_THRESHOLD = 1800;
+	const int NETDB_NUM_FLOODFILLS_THRESHOLD = 1800*4;
 	const int NETDB_NUM_ROUTERS_THRESHOLD = 4*NETDB_NUM_FLOODFILLS_THRESHOLD;
 	const int NETDB_TUNNEL_CREATION_RATE_THRESHOLD = 10; // in %
 	const int NETDB_CHECK_FOR_EXPIRATION_UPTIME = 600; // 10 minutes, in seconds  
-	const int NETDB_FLOODFILL_EXPIRATION_TIMEOUT = 60 * 60; // 1 hour, in seconds
-	const int NETDB_MIN_EXPIRATION_TIMEOUT = 90 * 60; // 1.5 hours
+	const int NETDB_FLOODFILL_EXPIRATION_TIMEOUT = 2 * 60 * 60; // 2 hours
+	const int NETDB_MIN_EXPIRATION_TIMEOUT = 2 * 60 * 60; // 2 hours
 	const int NETDB_MAX_EXPIRATION_TIMEOUT = 27 * 60 * 60; // 27 hours
 	const int NETDB_MAX_OFFLINE_EXPIRATION_TIMEOUT = 180; // in days
 	const int NETDB_EXPIRATION_TIMEOUT_THRESHOLD = 2*60; // 2 minutes
diff --git a/libi2pd/Streaming.cpp b/libi2pd/Streaming.cpp
index 25fb7b5..619281e 100644
--- a/libi2pd/Streaming.cpp
+++ b/libi2pd/Streaming.cpp
@@ -763,7 +763,7 @@ namespace stream
 					m_SlowRTT = m_MinRTT + m_Jitter;
 				}
 			}
-			if (m_IsBufferEmpty || m_FastRTT >= m_MinRTT + m_Jitter*3 || m_RTT >= m_MinRTT + m_Jitter*3 || m_SlowRTT >= m_MinRTT + m_Jitter*3 || m_RTT > m_FastRTT)
+			if (m_IsBufferEmpty || m_FastRTT >= m_MinRTT + m_Jitter*4 || m_RTT >= m_MinRTT + m_Jitter*4 || m_SlowRTT >= m_MinRTT + m_Jitter*4 || m_RTT > m_FastRTT)
 			{
 				incCounter = 0;
 				m_WindowIncCounter = 0;
@@ -771,7 +771,7 @@ namespace stream
 			m_WindowIncCounter = m_WindowIncCounter + incCounter;
 			//
 			// delay-based CC
-			if ((m_SlowRTT > m_MinRTT + m_Jitter*6) && !m_IsWinDropped && !m_IsClientChoked) // Drop window if RTT grows too fast
+			if ((m_SlowRTT > m_MinRTT + m_Jitter*8) && !m_IsWinDropped && !m_IsClientChoked) // Drop window if RTT grows too fast
 			{
 				LogPrint (eLogDebug, "Streaming: Congestion detected, reduce window size");
 				ProcessWindowDrop ();
diff --git a/libi2pd/Streaming.h b/libi2pd/Streaming.h
index d3bd314..0f3aa9c 100644
--- a/libi2pd/Streaming.h
+++ b/libi2pd/Streaming.h
@@ -59,10 +59,10 @@ namespace stream
 #endif	
 	const size_t COMPRESSION_THRESHOLD_SIZE = 66;
 	const int MAX_NUM_RESEND_ATTEMPTS = 10;
-	const int INITIAL_WINDOW_SIZE = 10;
-	const int MIN_WINDOW_SIZE = 3;
-	const int MAX_WINDOW_SIZE = 512;
-	const int MAX_WINDOW_SIZE_INC_PER_RTT = 12;
+	const int INITIAL_WINDOW_SIZE = 16;
+	const int MIN_WINDOW_SIZE = 5;
+	const int MAX_WINDOW_SIZE = 768;
+	const int MAX_WINDOW_SIZE_INC_PER_RTT = 768;
 	const double RTT_EWMA_ALPHA = 0.1;
 	const double SLOWRTT_EWMA_ALPHA = 0.02;
 	const double PREV_SPEED_KEEP_TIME_COEFF = 0.2; // 0.1 - 1 // how long will the window size stay around the previous drop level, less is longer
