diff --git a/libi2pd/Streaming.cpp b/libi2pd/Streaming.cpp
index 25fb7b5..2a0ead3 100644
--- a/libi2pd/Streaming.cpp
+++ b/libi2pd/Streaming.cpp
@@ -763,7 +763,7 @@ namespace stream
 					m_SlowRTT = m_MinRTT + m_Jitter;
 				}
 			}
-			if (m_IsBufferEmpty || m_FastRTT >= m_MinRTT + m_Jitter*3 || m_RTT >= m_MinRTT + m_Jitter*3 || m_SlowRTT >= m_MinRTT + m_Jitter*3 || m_RTT > m_FastRTT)
+			if (m_IsBufferEmpty || m_FastRTT >= m_MinRTT + m_Jitter*5 || m_RTT >= m_MinRTT + m_Jitter*5 || m_SlowRTT >= m_MinRTT + m_Jitter*5 || m_RTT > m_FastRTT)
 			{
 				incCounter = 0;
 				m_WindowIncCounter = 0;
@@ -771,7 +771,7 @@ namespace stream
 			m_WindowIncCounter = m_WindowIncCounter + incCounter;
 			//
 			// delay-based CC
-			if ((m_SlowRTT > m_MinRTT + m_Jitter*6) && !m_IsWinDropped && !m_IsClientChoked) // Drop window if RTT grows too fast
+			if ((m_SlowRTT > m_MinRTT + m_Jitter*10) && !m_IsWinDropped && !m_IsClientChoked) // Drop window if RTT grows too fast
 			{
 				LogPrint (eLogDebug, "Streaming: Congestion detected, reduce window size");
 				ProcessWindowDrop ();
diff --git a/libi2pd/Streaming.h b/libi2pd/Streaming.h
index d3bd314..8a84e06 100644
--- a/libi2pd/Streaming.h
+++ b/libi2pd/Streaming.h
@@ -59,10 +59,10 @@ namespace stream
 #endif	
 	const size_t COMPRESSION_THRESHOLD_SIZE = 66;
 	const int MAX_NUM_RESEND_ATTEMPTS = 10;
-	const int INITIAL_WINDOW_SIZE = 10;
-	const int MIN_WINDOW_SIZE = 3;
+	const int INITIAL_WINDOW_SIZE = 16;
+	const int MIN_WINDOW_SIZE = 5;
 	const int MAX_WINDOW_SIZE = 512;
-	const int MAX_WINDOW_SIZE_INC_PER_RTT = 12;
+	const int MAX_WINDOW_SIZE_INC_PER_RTT = 512;
 	const double RTT_EWMA_ALPHA = 0.1;
 	const double SLOWRTT_EWMA_ALPHA = 0.02;
 	const double PREV_SPEED_KEEP_TIME_COEFF = 0.2; // 0.1 - 1 // how long will the window size stay around the previous drop level, less is longer
